File: App1/HandPoint/HandPointSDKWrapper.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Com.Handpoint.Api;
using Com.Handpoint.Api.Shared;
using Java.Interop;
using JustTouchMobile.Services.Payments;
using System;
using System.Collections.Generic;
using Application = Android.App.Application;

namespace JustTouchMobile.Droid.HandPoint {
	class HandPointSDKWrapper : Java.Lang.Object, IPayments, Com.Handpoint.Api.Shared.Events.IRequired, Com.Handpoint.Api.Shared.Events.IConnectionStatusChanged, Com.Handpoint.Api.Shared.Events.ICurrentTransactionStatus {
		
		private Hapi api;
        private Context mContext;

        public HandPointSDKWrapper(Context context) {
            mContext = context;
        }

        public void AddEventHandler(System.EventHandler eventHandler) {
            throw new NotImplementedException();
        }
		public void RemoveEventHandler(System.EventHandler eventHandler) {
			throw new NotImplementedException();
		}

		public void InitPayment(string sharedKey) {

            //Please reference same code here https://www.handpoint.com/docs/device/Android/6.1.0/

            var cloudKey = "M9KC2NP-632MZZZ-KH6EMM5-WVMYYTM";

            HandpointCredentials handpointCredentials = new HandpointCredentials(sharedKey);

            try
            {
                this.api = HapiFactory.GetAsyncInterface(this, mContext, handpointCredentials);
                this.api.SetLogLevel(LogLevel.Debug);

                // this is connect directly to to device
                Device device = new Device("LocalDevice", "0821638950-PAXA920", "", ConnectionMethod.AndroidPayment);
                // this is to connect to the handpoint payment app via the cloud
                //Device device = new Device("Cloud Device", "0821638950-PAXA920", "", ConnectionMethod.Cloud);
                
                this.api.Connect(device);

            }
            catch (Exception ex) {
                Console.WriteLine(ex);
            }
		}

		public bool Print(string html) {
			return this.api.PrintReceipt (html);
		}

		public bool Refund(decimal amount) {
			throw new NotImplementedException();
			//this.api.Refund();
		}


        public bool Sale(decimal amount) {

            if (this.api != null)
            {
                Java.Math.BigInteger bigInteger = new Java.Math.BigInteger(amount.ToString());
                return this.api.Sale(bigInteger, Currency.Usd);
            }

            return false;
		}

        public void DeviceDiscoveryFinished(IList<Device> devices) {

        }

        public void EndOfTransaction(TransactionResult result, Device device) {
            
        }

        public void TransactionResultReady(TransactionResult transactionResult, Device device) {
            
        }

        public void SignatureRequired(SignatureRequest request, Device device) {
           
        }

        public void ConnectionStatusChanged(ConnectionStatus status, Device device) {
            Console.WriteLine($"ConnectionStatusChanged - {status}");
            
            if (status == ConnectionStatus.Connected) {
                Sale(100);
            }

        }

        public void CurrentTransactionStatus(StatusInfo info, Device device) {
            Console.WriteLine($"CurrentTransactionStatus - {info}");

        }
    }
}

File: Java/MyApplication/app/src/main/java/com/example/myapplication/HandPointSDKWrapper.java
package com.example.myapplication;

import android.content.Context;

import com.handpoint.api.HandpointCredentials;
import com.handpoint.api.Hapi;
import com.handpoint.api.HapiFactory;
import com.handpoint.api.shared.ConnectionMethod;
import com.handpoint.api.shared.ConnectionStatus;
import com.handpoint.api.shared.Device;
import com.handpoint.api.shared.Events;
import com.handpoint.api.shared.SignatureRequest;
import com.handpoint.api.shared.StatusInfo;
import com.handpoint.api.shared.TransactionResult;

import java.util.List;

public class HandPointSDKWrapper implements Events.Required, Events.ConnectionStatusChanged, Events.CurrentTransactionStatus {

    private Hapi api;

    public HandPointSDKWrapper(Context context) {
        initApi(context);
    }

    public void initApi(Context context) {
        String sharedSecret = "5FC9200DEE75F8AB2D37B6D74E7ECB075EEA69625DCFCEC37A9A42670E16960B";
        HandpointCredentials handpointCredentials = new HandpointCredentials(sharedSecret);
        this.api = HapiFactory.getAsyncInterface(this, context, handpointCredentials);
        // The api is now initialized. Yay! we've even set default credentials.
        // The shared secret is a unique string shared between the payment terminal and your application, it is unique per merchant.
        // You should replace this default shared secret with the one sent by the Handpoint support team.
        //Since we're running inside the terminal, we can create a device ourselves and connect to it
        Device device = new Device("some name", "address", "", ConnectionMethod.ANDROID_PAYMENT);
        this.api.connect(device);
    }

    @Override
    public void connectionStatusChanged(ConnectionStatus connectionStatus, Device device) {

    }

    @Override
    public void currentTransactionStatus(StatusInfo statusInfo, Device device) {

    }

    @Override
    public void deviceDiscoveryFinished(List<Device> list) {

    }

    @Override
    public void endOfTransaction(TransactionResult transactionResult, Device device) {

    }

    @Override
    public void signatureRequired(SignatureRequest signatureRequest, Device device) {

    }

    @Override
    public void transactionResultReady(TransactionResult transactionResult, Device device) {

    }
}
